src/framework/preferences/js/SyllabificationEnactor.js

```diff
@@ -218,29 +215,74 @@ var fluid_3_0_0 = fluid_3_0_0 || {};
         var hyphenator = fluid.getGlobalValue(globalPath);
 
         // If the pattern file has already been loaded, return the hyphenator.
+        // This could happen if the pattern file is statically linked to the page.
         if (hyphenator) {
             promise.resolve(hyphenator);
             return promise;
         }
 
         var src = fluid.stringTemplate(pattern, that.options.terms);
+
         var injectPromise = that.injectScript(src);
         injectPromise.then(function () {
             hyphenator = fluid.getGlobalValue(globalPath);
             promise.resolve(hyphenator);
-        }, function (jqXHR, textStatus, errorThrown) {
-            that.events.onError.fire(
-                "The pattern file \"" + src + "\" could not be loaded.",
-                jqXHR,
-                textStatus,
-                errorThrown
-            );
-            // If the pattern file could not be loaded resolve the promise with out a hyphenator (undefined).
+        }, function () {
+            var errorMessage = fluid.stringTemplate(that.options.strings.patternLoadError, {src: src});
+            // Pass along arguments fired for the rejection, but with our errorMessage as the first argument
+            var args = [errorMessage].concat(fluid.values(arguments));
```

Comment:
fluid.values is an odd choice given arguments is arraylike - we would typically use fluid.makeArray here. But in fact there is no choice about the signature - the promise API is fixed and always provides a single-argument error payload. So you can just say function (error) and push that into the argument list. As a courtesy it might be worth checking whether it is a string and if so simply appending it to the message rather than dangling it off the argument list where it might not be seen (especially since this argument list is not documented). Another useful courtesy might be to send the full error message to fluid.log here in addition to firing the event.